pkg/proxy/proxier.go

```diff
@@ -368,12 +381,80 @@ func (proxier *Proxier) openOnePortal(portalIP net.IP, portalPort int, protocol
 	return nil
 }
 
+func (proxier *Proxier) lockPort(port int, owner ServicePortName) error {
+	proxier.portMapMutex.Lock()
+	defer proxier.portMapMutex.Unlock()
+
+	// TODO: We could pre-populate some reserved ports into portMap and/or blacklist some well-known ports
+	existing, found := proxier.portMap[port]
+	if !found {
+		proxier.portMap[port] = owner
+		return nil
+	}
+	if existing == owner {
```

Comment:
We sync the rules periodically.

SyncLoop -> ensurePortals -> openPortal -> openNodePort -> claimPort